Makefile

```diff
@@ -68,22 +68,24 @@ CUFINUFFTOBJS_64=src/2d/spreadinterp2d.o src/2d/cufinufft2d.o \
 	src/3d/interp3d_wrapper.o src/3d/cufinufft3d.o \
 	$(CONTRIBOBJS)
 CUFINUFFTOBJS_32=$(CUFINUFFTOBJS_64:%.o=%_32.o)
+$(info $$CUFINUFFTOBJS_32 is [${CUFINUFFTOBJS_32}])
+
 
 CUFINUFFTCOBJS_64=src/cufinufftc.o
 CUFINUFFTCOBJS_32=$(CUFINUFFTCOBJS_64:%.o=%_32.o)
 
-%.o: %.cpp $(HEADERS)
-	$(CXX) -c $(CXXFLAGS) $(INC) $< -o $@
-%.o: %.c $(HEADERS)
-	$(CC) -c $(CFLAGS) $(INC) $< -o $@
-%.o: %.cu $(HEADERS)
-	$(NVCC) --device-c -c $(NVCCFLAGS) $(INC) $< -o $@
 %_32.o: %.cpp $(HEADERS)
 	$(CXX) -DSINGLE -c $(CXXFLAGS) $(INC) $< -o $@
 %_32.o: %.c $(HEADERS)
 	$(CC) -DSINGLE -c $(CFLAGS) $(INC) $< -o $@
 %_32.o: %.cu $(HEADERS)
 	$(NVCC) -DSINGLE --device-c -c $(NVCCFLAGS) $(INC) $< -o $@
+%.o: %.cpp $(HEADERS)
```

Comment:
note, I had to move this and some other of the more specific patterns to precede the more generic patterns for old systems (like manylinux2010's based on centos6).  I think is it because they are using an ancient gnu make (3.8) which may have slightly different rule evaluation, but I never found the exact documentation about what changed (nor did I look, I've seen this before).

Regardless, this order works for everything I have thrown at it.